### PR TITLE
Add concurrency to the find-large-objects scrubber subcommand

### PR DESCRIPTION
**Pull request by arpad-m**
The find-large-objects scrubber subcommand is quite fast if you run it in an environment with low latency to the S3 bucket (say an EC2 instance in the same region). However, the higher the latency gets, the slower the command becomes. Therefore, add a concurrency param and make it parallelized. This doesn't change that general relationship, but at least lets us do multiple requests in parallel and therefore hopefully faster.

Running with concurrency of 64 (default):

```
2024-07-05T17:30:22.882959Z  INFO lazy_load_identity [...]
[...]
2024-07-05T17:30:28.289853Z  INFO Scanned 500 shards. [...]
```

With concurrency of 1, simulating state before this PR:

```
2024-07-05T17:31:43.375153Z  INFO lazy_load_identity [...]
[...]
2024-07-05T17:33:51.987092Z  INFO Scanned 500 shards. [...]
```

In other words, to list 500 shards, speed is increased from 2:08 minutes to 6 seconds.

Follow-up of  #8257, part of #5431

### CODE REVIEW

File: storage_scrubber/src/find_large_objects.rs [hunk 1]
```diff
@@ -45,53 +50,85 @@ pub async fn find_large_objects(
...
             }
+
+            tenant_ctr.fetch_add(1, Ordering::Relaxed);
+
+            Ok((tenant_shard_id, objects))
         }
+    });
+    let mut objects_stream = std::pin::pin!(objects_stream.try_buffer_unordered(concurrency));
+
+    //let mut objects_stream = objects_stream.flatten();
```
Comment (skyzh) [inline storage_scrubber/src/find_large_objects.rs:108]: rm comments?

File: storage_scrubber/src/find_large_objects.rs [hunk 2]
```diff
@@ -45,53 +50,85 @@ pub async fn find_large_objects(
...
+
+    let mut objects = Vec::new();
+    while let Some(res) = objects_stream.next().await {
```
Comment (skyzh) [inline storage_scrubber/src/find_large_objects.rs:111]: does this really increase speed? every `next` waits until a tenant finishes before continuing.

I think to achieve true concurrency, we need either `tokio::spawn` to create new tasks, or `futures_unordered` to poll multiple futures in a future.

Comment (skyzh) [reply storage_scrubber/src/find_large_objects.rs:111]: sorry I didn't see try_buffer_unordered

**Review by skyzh — APPROVED** [review]
LGTM

**Comment by skyzh** [comment]
created a tracking issue for test lsn lease flaky: https://github.com/neondatabase/neon/issues/8293

File: storage_scrubber/src/find_large_objects.rs (hunk 1 above)
Comment (arpad-m) [reply storage_scrubber/src/find_large_objects.rs]: done.
